src/components/Web3Provider/index.test.tsx

```diff
@@ -0,0 +1,48 @@
+import { act, render } from '@testing-library/react'
+import { initializeConnector, MockEIP1193Provider } from '@web3-react/core'
+import { EIP1193 } from '@web3-react/eip1193'
+import { Provider as EIP1193Provider } from '@web3-react/types'
+import { Connection, ConnectionType } from 'connection'
+import useEagerlyConnect from 'hooks/useEagerlyConnect'
+import useOrderedConnections from 'hooks/useOrderedConnections'
+import { Provider } from 'react-redux'
+import store from 'state'
+import { asMock } from 'test-utils'
+
+import Web3Provider from '.'
+
+jest.mock('hooks/useEagerlyConnect', () => jest.fn())
+jest.mock('hooks/useOrderedConnections', () => jest.fn())
+
+const UI = (
+  <Provider store={store}>
+    <Web3Provider>{null}</Web3Provider>
+  </Provider>
+)
+
+describe('Web3Provider', () => {
+  let provider: MockEIP1193Provider & EIP1193Provider
+  let connection: Connection
+
+  beforeEach(() => {
+    provider = new MockEIP1193Provider() as MockEIP1193Provider & EIP1193Provider
+    const [connector, hooks] = initializeConnector((actions) => new EIP1193({ actions, provider }))
+    connection = {
+      connector,
+      hooks,
+      getName: jest.fn().mockReturnValue('test'),
+      type: 'INJECTED' as ConnectionType,
+      shouldDisplay: () => false,
+    }
+    asMock(useOrderedConnections).mockReturnValue([connection])
+  })
+
+  it('renders', async () => {
```

Comment:
idk about test name convention but this looks like to me like this is testing that the `Web3Provider` `renders and eagerly connects`. and i feel like for something like this without a UI impact, we do actually want to test more for this connecting behavior

Reply:
This is just a simple test for `render`, it is _not_ a test for `useEagerlyConnect`. That would need to be tested separately, as it is specific to the connectors used in interface and will not function with the simple mocked connection used for the render test here.

This is just to lay a foundation of testing for this file. It is not meant to be exhaustive.

Reply:
i get mocking something if it needs to be for it to be rendered, but the fact that you have `expect(useEagerlyConnect).toHaveBeenCalled()` means that you expect this component to call the hook, which means you're also testing more than just rendering. so technically i'd say you should either leave out that expect to have been called from this test, add the `and calls eagerly connect` to the description for this test, or have a separate test for render and this hook being called.